📝 ProviderState (#3)
DimensionDev/Mask-Docs: docs/web3/provider.md

```

In Mask, we provide hooks to access them.

```ts
const web3 = useWeb3(NetworkPluginID.PLUGIN_EVM)
const web3Provider = useWeb3Provider(NetworkPluginID.PLUGIN_EVM)
```

Integrating a wallet provider should implement the `WalletProvider` interface. We assume there is lazy progress that a wallet provider will need to do the preparation, and it can switch to a specific sub-network. It also creates an SDK instance and a provider instance that implements the provider protocol of that network.

```ts
export interface WalletProvider<ChainId, ProviderType, Web3Provider, Web3> {
    /** Keep waiting until the provider is ready. */
    readonly readyPromise: Promise<void>
    /** Switch to the designate chain. */
    switchChain(chainId?: ChainId): Promise<void>
    /** Create an instance that creates from the network SDK. */
    createWeb3(chainId?: ChainId): Promise<Web3>
    /** Create an instance that implement the wallet protocol. */
    createWeb3Provider(chainId?: ChainId): Promise<Web3Provider>
    /** Create the connection. */
    connect(chainId?: ChainId): Promise<Account<ChainId>>
    /** Dismiss the connection. */
    disconnect(): Promise<void>
}
```

> The `ProviderState` takes mangement of multiple wallet provoders. It tracks the currently selected account address and which sub-network is the account stays on. Also, we access them with hooks.

Author: ```suggestion
The `ProviderState` takes management of multiple wallet providers. It tracks the currently selected account address and in which sub-network the account stays. Of course, we can access them with hooks.
```
